### PR TITLE
add SecuredScriptureRangesOverride and option to SecuredPropertyList

### PR DESCRIPTION
**Pull request by tjgersho**

### CODE REVIEW

File: src/components/product/dto/product.dto.ts [hunk 1]
```diff
@@ -56,7 +56,7 @@ export class DerivativeScriptureProduct extends Product {
       this property can be set (and read) to "override" the \`producible\`'s list.
     `,
   })
-  readonly scriptureReferencesOverride?: SecuredScriptureRanges;
+  readonly scriptureReferencesOverride?: SecuredScriptureRangesOverride;
```
Comment (CarsonF) [inline src/components/product/dto/product.dto.ts:59]: This field should no longer be nullable

File: src/common/secured-property.ts [hunk 2]
```diff
@@ -50,13 +50,20 @@ SecuredProperty.descriptionFor = (value: string) => stripIndent`
...
-  valueClass: Class<T> | AbstractClassType<T> | GraphQLScalarType | object
+  valueClass: Class<T> | AbstractClassType<T> | GraphQLScalarType | object,
+  options: OverrideOption = {}
 ) {
   @ObjectType({ isAbstract: true, implements: [Readable, Editable] })
   abstract class SecuredPropertyListClass
     implements Readable, Editable, Secured<T[]> {
-    @Field(() => [valueClass])
+    @Field(() => [valueClass], {
+      nullable: options.isOverride ? options.isOverride : false,
```
Comment (CarsonF) [inline src/common/secured-property.ts:65]:
```suggestion
      nullable: options.isOverride,
```

File: src/common/secured-property.ts [hunk 3]
```diff
@@ -50,13 +50,20 @@ SecuredProperty.descriptionFor = (value: string) => stripIndent`
   These \`can*\` authorization properties are specific to the user making the request.
 `;
 
+export interface OverrideOption {
```
Comment (CarsonF) [inline src/common/secured-property.ts:53]:
```suggestion
export interface SecuredPropertyListOptions {
```

File: src/components/product/dto/product.dto.ts (hunk 1 above)
Comment (CarsonF) [reply src/components/product/dto/product.dto.ts:58]:
```suggestion
  readonly scriptureReferencesOverride: SecuredScriptureRangesOverride;
```

Comment (CarsonF) [reply src/components/product/dto/product.dto.ts:58]: Ok I just removed the `?` myself since @michaelmarshall merged it.

**Comment by CarsonF** [comment]
@tjgersho I pushed the removal of `?` to master. But that broke master. Can you fix that and open another PR?

**Comment by tjgersho** [comment]
Sure.

**Pull request merged**